src/hex_tarball.erl

```diff
@@ -54,9 +56,11 @@ create(Metadata, Files) ->
        {"contents.tar.gz", ContentsBinary}
     ],
 
-    Tarball = create_tarball(OuterFiles, []),
+    Tarball = build_tarball(OuterFiles),
 
-    case byte_size(Tarball) > ?TARBALL_MAX_SIZE of
+    UncompressedSize = byte_size(Tarball) - byte_size(ContentsBinary) + byte_size(ContentsBinaryBuild),
```

Comment:
I think it can simply be: `UncompressedSize = byte_size(ContentsBinaryBuild)`, I think we can ignore other outer files' sizes.